README.md

```diff
@@ -1 +1,98 @@
 # Bauble
+
+Bauble is a data format targeted at games with rust-like syntax.
+
+It is intended to be used with a system that has some kind of reflection on types. So types can be optionally included instead of refering to them by full path. And it also gives the data-format the potential to load erased types.
+
+Bauble has three steps in parsing.
+1. text -> simple intermediate data representation. 
+2. -> intermediate data representation with resolved types.
+3. -> rust types through implementation of `FromBauble`. 
+
+## Examples of bauble
+```rust
+// bauble has the capability to use different types. Which are resolved by the `AssetCtx`
+use rpg::{Enemy, DamageType};
+
+slime = Enemy {
+  name: "Slime",
+  hp: 200,
+  speed: 2.5,
+  damage_type: DamageType::Slimy,
+  // Arrays, which could be deserialized as other containers, are deliminated by `[]`
+  attacks: [$slam_attack],
+}
+
+// Putting `copy` infront of a definition essentially makes it a macro.
+// As in rust, tuples are delimeted by `()`.
+copy slam_attack = (2.5, "Slam")
+```
+
+Bauble also supports a raw piece of data, that could potentially be used for code. Raw values need to be defined how to be handled when they are converted to rust types.
+```rust
+use rpg::{Ability, Projectile, OnUse};
+fireball = Ability {
+  name: "Fireball",
+  // You can link to assets by prefixing a path with `$`.
+  audio: $assets::sounds::fireball_audio,
+  // You can link to other assets within the document.
+  on_use: OnUse::SpawnProjectile($fireball_projectile),
+}
+
+fireball_projectile = Projectile {
+  // After a `#` a raw value is accepted. This could be used for a color.
+  color: #42f56f,
+  // Or a dice roll.
+  hit_damage: #2d6,
+  // Everything within #{ ... } is considered raw data. Other instances of `{` must be delimited by `}`.
+  on_hit: #{
+    ctx.spawn_effect("FireballEffect");
+    ctx.play_sound("FireballSound");
+    for enemy in ctx.nearby_enemies(1.0) {
+      enemy.damage(ctx.data.hit_damage);
+    }
+  }
+}
+```
+
+For just a raw value after `#`, the following characters are accepted:
+ - Special char acters: `!`, `#`, `@`, `%`, `&`, `?`, `.`, `=`, `<`, `>`, `_`, `-`, `+`, `*`
+ - Alphabetical numerical characters, see (is_alphanumeric)[https://doc.rust-lang.org/std/primitive.char.html#method.is_alphanumeric].
+
+There is no specific reason this list can't be expanded.
```

Comment:
Maybe simplify to "This list can potentially be expanded". 

(Though btw, is this list hard-coded into the library? What would the ways of expanding this be? Updating the library? Are we talking about someone proposing adding a character because it makes sense, or that people can just themselves expand upon this list as library users)